Source/Core/Core/FifoPlayer/FifoRecorder.h

```diff
@@ -48,6 +50,37 @@ class FifoRecorder
   static FifoRecorder& GetInstance();
 
 private:
+  class FifoRecordAnalyzer : public OpcodeDecoder::Callback
+  {
+  public:
+    FifoRecordAnalyzer(FifoRecorder* owner) : m_owner(owner) {}
+    FifoRecordAnalyzer(FifoRecorder* owner, const u32* cpmem) : m_owner(owner), m_cpmem(cpmem) {}
+
+    void OnXF(u16 address, u8 count, const u8* data) override {}
+    void OnCP(u8 command, u32 value) override {}
+    void OnBP(u8 command, u32 value) override {}
+    void OnIndexedLoad(u8 array, u32 index, u16 address, u8 size) override;
+    void OnPrimitiveCommand(OpcodeDecoder::Primitive primitive, u8 vat, u32 vertex_size,
+                            u16 num_vertices, const u8* vertex_data) override;
+    void OnDisplayList(u32 address, u32 size) override
+    {
+      // Should have been inlined by the recorder
+      ASSERT(false);
```

Comment:
Maybe implement this in the cpp file, so the assert header doesn't need to be included in another header.

Reply:
I've moved `FifoRecordAnalyzer` to the cpp file; however, I also moved the main OpcodeDecoding logic into the header for inlining, which includes other asserts (that are probably more useful).

Reply:
Oh, I should add: I replaced *this* assert with a warning log, as it's in the recording code, which currently assumes that display lists will be inlined so one should never reach here.  But I did remove the asserts on analysis and playback.